### PR TITLE
Consolidate lint files

### PR DESCRIPTION
**Pull request by jpsim**
This is setting up for a dedicated autocorrect command, and is a lot nicer to work with IMO

### CODE REVIEW

File: Source/swiftlint/Extensions/Configuration+CommandLine.swift [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    func getFiles(path: String, action: String, useSTDIN: Bool) ->
+                  Result<[File], CommandantError<()>> {
+        if useSTDIN {
+            let standardInput = NSFileHandle.fileHandleWithStandardInput()
+            let stdinData = standardInput.readDataToEndOfFile()
+            let stdinNSString = NSString(data: stdinData, encoding: NSUTF8StringEncoding)
+            if let stdinString = stdinNSString as? String {
+                return .Success([File(contents: stdinString)])
+            }
+            return .Failure(.UsageError(description: "stdin isn't a string"))
```
Comment (segiddins) [inline Source/swiftlint/Extensions/Configuration+CommandLine.swift]: might want to mention UTF-8 in this error

**Comment by jpsim** [comment]
Ok, I ended up making a lot more changes in here because it would have been too much work to split it up. Specifically a `Configuration.visitLintableFiles(...)` visitor function which will more easily allow performing other operations on the same lintable files (e.g. autocorrecting).

File: Source/swiftlint/Extensions/Configuration+CommandLine.swift (hunk 1 above)
Comment (jpsim) [inline Source/swiftlint/Extensions/Configuration+CommandLine.swift]: thanks for the tip, done

**Comment by jpsim** [comment]
Since this is just an internal refactor, I'll merge this one now. @keith if you have feedback, I'll do my best to incorporate it afterwards.

**Pull request merged**